Statements issue (#2073)
Codecademy/docs: content/javascript/concepts/statements/statements.md

---
Title: 'Statements'
Description: 'In JavaScript, a statement is a unit of code that performs a specific action or task.'
Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Statements'
  - 'Syntax'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

In JavaScript, **statements** are instructions that are executed by the browser or Node.js. A statement can be a simple or complex operation that performs an action, such as assigning a value to a variable, calling a function, or controlling program flow with conditional statements.

> ## Syntax

Reviewer: I do appreciate this syntax section, however I think this entry would do better with breaking up statements into the types of statements and then you can go into the syntax within each section. You can set the first h2 to ## Declaration

> In this statement, a const variable called `greeting` is declared and assigned the value of `'Hello World'`:

Reviewer: ```suggestion
In this statement, a `const` variable named `greeting` is declared and assigned the value of `'Hello World'`:
```

```js
const greeting = 'Hello World';
```

The first statement below assigns the value 5 to the variable `x`, while the second statement logs the value of `x` to the console:

```js
let x = 5;
console.log(x);
```

JavaScript variables are containers for storing data values.
In this example, `x`, `y`, and `z`, are variables, declared with the `var` keyword:

```js
var x = 5;
var y = 6;
var z = x + y;
```

JavaScript identifiers are case-sensitive. The names `x` and `X` have different meanings:

```js
// Assign 5 to x
var x = 5;
// Assign 6 to X
var X = 6;
```

## Types of Statements

There are several types of statements in JavaScript, including:

> **Expression** statements: statements that evaluate an expression, such as function calls or assignments:

Reviewer: I don't believe these are expressions. If we declared the variable, then the next time it's incremented  like with `x++` or something of that sort, that would be considered an expression, but I don't believe the initial declaration and assignment is.

Please link me to your source so we can confirm.

Author: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators

to clarify, would it be correct to do this:
let x = 5 + 5;

as this would be an expression?

Reviewer: Thanks for linking that! Good doc to reference.

Looks like what you wrote in the syntax section was accurate and will fit nicely in an Expression Statements section. From the documentation, it looks like a declaration `let x` along with an assignment (some value) is considered an expression. And yes, your example `let x = 5 + 5` is also an expression.

Reviewer: Then I suggest shifting this syntax section as an h3 underneath ## Expression Statements

```js
// Function call
console.log('Hello World');
// Assignment
let x = 5;
```

> **Conditional** statements: statements that control program flow based on a condition, such as `if` statements or `switch` statements:

Reviewer: It would be a good idea to turn this into its own header (no need to bold)

Author: does this mean I would need to do this on each statement type?

---
Title: 'Statements'
Description: 'In JavaScript, a statement is a unit of code that performs a specific action or task.'
Subjects:
 - 'Web Development'
 - 'Computer Science'
Tags:

>  - 'Statements'

Reviewer 2: ```suggestion
 - 'Statements'
 - 'Syntax'
```

CatalogContent:
 - 'introduction-to-javascript'
 - 'paths/front-end-engineer-career-path'
---

In JavaScript, statements are instructions that are executed by the browser or Node.js. A statement can be a simple or complex operation that performs an action, such as assigning a value to a variable, calling a function, or controlling program flow with conditional statements.

## Syntax

> In this statement, a const variable called greeting is declared and assigned the value of 'Hello World':

Reviewer 2: ```suggestion
In this statement, a const variable called `greeting` is declared and assigned the value of `'Hello World'`:
```

```js
const greeting = 'Hello World';
```

> The first statement assigns the value 5 to the variable x, while the second statement logs the value of x to the console:

Reviewer 2: ```suggestion
The first statement below assigns the value 5 to the variable `x`, while the second statement logs the value of `x` to the console:
```

```js
let x = 5;
console.log(x);
```

JavaScript variables are containers for storing data values.

> In this example, x, y, and z, are variables, declared with the var keyword:

Reviewer 2: ```suggestion
In this example, `x`, `y`, and `z`, are variables, declared with the `var` keyword:
```

    
```js
var x = 5;
var y = 6;
var z = x + y;
``` 

> JavaScript identifiers are case-sensitive. The names x and X have different meanings:

Reviewer 2: ```suggestion
JavaScript identifiers are case-sensitive. The names `x` and `X` have different meanings:
```

There are several types of statements in JavaScript, including:

Expression statements: statements that evaluate an expression, such as function calls or assignments:

```js
// Function call
console.log('Hello World');
// Assignment
let x = 5;
```
Conditional statements: statements that control program flow based on a condition, such as if statements or switch statements:

```js
// If statement
if (x > 5) {
  console.log('x is greater than 5');
}
// Switch statement
switch (x) {
  case 1:
    console.log('x is 1');
    break;
  case 2:
    console.log('x is 2');
    break;
  default:
    console.log('x is neither 1 nor 2');
}
```

Loop statements: statements that repeat an operation while a condition is true, such as while loops or for loops:

```js
// While loop
while (x < 5) {
  console.log(x);
  x++;
}
// For loop
for (let i = 0; i < 5; i++) {
  console.log(i);
}
```

Jump statements: statements that transfer control to another part of the program, such as break or return statements:

```js
// Break statement
for (let i = 0; i < 5; i++) {
  if (i === 3) {
    break;
  }
  console.log(i);
}
// Return statement
function add(x, y) {
  return x + y;
}
```
## Notes

> To summarize, JavaScript statements are instructions that tell the browser what to do.

Reviewer 2: I don't believe this is adding much to this entry, please delete.

---
Title: 'Statements'
Description: 'In JavaScript, a statement is a unit of code that performs a specific action or task.'
Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Statements'
  - 'Syntax'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

In JavaScript, **statements** are instructions that are executed by the browser or Node.js. A statement can be a simple or complex operation that performs an action, such as assigning a value to a variable, calling a function, or controlling program flow with conditional statements.

> ## Expression Statements
> 
> Expression statements, such as function calls or assignments, evaluate an expression and generate a value in JavaScript. They can be assigned to a [variable](https://www.codecademy.com/resources/docs/javascript/variables) or used as part of a larger expression.
> 
> ### Syntax
> 
> ```js
> // Function call
> console.log('Hello World');
> // Assignments
> // Assign 5 to x
> var x = 5;
> // Assign 6 to X
> var X = 6;
> ```
> 
> In this statement, a `const` variable named `greeting` is declared and assigned the value of `'Hello World'`:
> 
> ### Syntax
> 
> ```js
> const greeting = 'Hello World';
> ```
> 
> The first statement below assigns the value 5 to the variable `x`, while the second statement logs the value of `x` to the console:
> 
> ### Syntax
> 
> ```js
> let x = 5;
> console.log(x);
> ```
> 
> JavaScript [variables](https://www.codecademy.com/resources/docs/javascript/variables) are containers for storing data values.
> In this example, `x`, `y`, `X`, and `z`, are [variables](https://www.codecademy.com/resources/docs/javascript/variables), declared with the `var` keyword.
> JavaScript identifiers are also case-sensitive. Note that `x` is not the same as `X`:
> 
> ### Syntax
> 
> ```js
> var x = 5;
> var y = 6;
> var X = 4;
> var z = x + y;
> ```

Reviewer: ```suggestion
## Declaration Statements

Declaration statements are variables, function, or classes that are introduced into a program. These statements begin with a keyword followed by its identifier or name.

```js
var x = 5;
var y = 6;
var X = 4;
var z = x + y;
```

JavaScript [variables](https://www.codecademy.com/resources/docs/javascript/variables) are containers for storing data values. In this example, `x`, `y`, `X`, and `z`, are declared with the `var` keyword.

JavaScript identifiers are also case-sensitive (Note that `x` is not the same as `X`).

## Expression Statements

Expression statements, such as function calls or assignments, evaluate an expression and generate a value in JavaScript. They can be assigned to a variable or used as part of a larger expression and are then discarded.

```js
// Function call
console.log('Hello World');
```

The second statement below re-assigns the value 5 to the variable `x`, while the third statement logs the value of `x` to the console:

```js
var x = 5;
x = 8;
console.log(x);
```
```

> ```

Reviewer: @nadocodes Of course! What's most important is understanding and filling in any gaps or misunderstandings. Which is a great outlook because I made an error here.

I tried putting in a suggestion but there's some formatting issues which won't let me put it in as a suggestion from lines 18-62 because of the early suggestion block's closing backticks. This is what I recommend:

```
## Declaration Statements

Declaration statements are variables, function, or classes that are introduced into a program. These statements begin with a keyword followed by its identifier or name.

```js
var x = 5;
var y = 6;
var X = 4;
var z = x + y;
```

JavaScript [variables](https://www.codecademy.com/resources/docs/javascript/variables) are containers for storing data values. In this example, `x`, `y`, `X`, and `z`, are declared with the `var` keyword.

JavaScript identifiers are also case-sensitive (Note that `x` is not the same as `X`).

## Expression Statements

Expression statements, such as function calls or assignments, evaluate an expression and generate a value in JavaScript. They can be assigned to a variable or used as part of a larger expression and are then discarded.

```js
// Function call
console.log('Hello World');
```

The second statement below re-assigns the value 5 to the variable `x`, while the third statement logs the value of `x` to the console:

```js
var x = 5;
x = 8;
console.log(x);
```
```

Reviewer: Our original understanding of declaration statements v expression statements was in fact technically accurate. The Mozilla docs tripped me up

Author: Perfect!! I have made the changes and hoping I havnt missed anything, I am wondering if I should leave the "const greeting = 'Hello World'; to highlight varied variable use?

Author: Thank you 😄

## Conditional Statements

[Conditional](https://www.codecademy.com/resources/docs/javascript/conditionals) statements, such as `if` statements or `switch` statements, control program flow based on a condition.

> 
> ### Syntax

Reviewer: ```suggestion
```

Looking over this and now seeing the repeated headers isn't very flattering to the eye.

```js
// If statement
if (x > 5) {
  console.log('x is greater than 5');
}
// Switch statement
switch (x) {
  case 1:
    console.log('x is 1');
    break;
  case 2:
    console.log('x is 2');
    break;
  default:
    console.log('x is neither 1 nor 2');
}
```

## Loop Statements

[Loop](https://www.codecademy.com/resources/docs/javascript/loops) statements, such as `while` loops or `for` loops, repeat an operation while a condition is true.

> 
> ### Syntax

Reviewer: ```suggestion
```

```js
// While loop
while (x < 5) {
  console.log(x);
  x++;
}
// For loop
for (let i = 0; i < 5; i++) {
  console.log(i);
}
```

## Jump Statements

Jump statements, such as `break` or `return` statements, transfer control to another part of the program.

> 
> ### Syntax

Reviewer: ```suggestion
```

Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Statements'
  - 'Syntax'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

In JavaScript, **statements** are instructions that are executed by the browser or Node.js. A statement can be a simple or complex operation that performs an action, such as assigning a value to a variable, calling a function, or controlling program flow with conditional statements.

## Declaration

In this statement, a `const` variable named `greeting` is declared and assigned the value of `'Hello World'`:

### Syntax

```js
const greeting = 'Hello World';
```

The first statement below assigns the value 5 to the variable `x`, while the second statement logs the value of `x` to the console:

```js
let x = 5;
console.log(x);
```

JavaScript variables are containers for storing data values.
In this example, `x`, `y`, `X`, and `z`, are variables, declared with the `var` keyword.
JavaScript identifiers are also case-sensitive. Note that `x` is not the same as `X`:

```js
var x = 5;
var y = 6;
var X = 4;
var z = x + y;

> ```

Reviewer: There may have been some confusion from my previous response. Can we move this to the syntax section underneath "Expression Statements" instead. Also, feel free to just use either var or let.

Author: Hi, sorry about the confusion, I am currently doing as you suggested, I'm wondering if I should move the entire Declaration section over to Expressions Statements? Again sorry :/

Author: Please check my changes, I hope I am on track!

## Expression Statements

Statements that evaluate an expression, such as function calls or assignments:

### Syntax

```js
// Function call
console.log('Hello World');
// Assignment
let x = 5;
// Assign 5 to x

> var x = 5;

Reviewer: This is actually not allowed once you've declared x with let. I would just switch this syntax section with the one above.

Reviewer: I also recommend (when mentioning variables) to link to https://www.codecademy.com/resources/docs/javascript/variables

// Assign 6 to X
var X = 6;
```

## Conditional Statements

> Statements that control program flow based on a condition, such as `if` statements or `switch` statements:

Reviewer: ```suggestion
[Conditionals](https://www.codecademy.com/resources/docs/javascript/conditionals), such as `if` statements or `switch` statements, control program flow based on a condition.
```

### Syntax

```js
// If statement
if (x > 5) {
  console.log('x is greater than 5');
}
// Switch statement
switch (x) {
  case 1:
    console.log('x is 1');
    break;
  case 2:
    console.log('x is 2');
    break;
  default:
    console.log('x is neither 1 nor 2');
}
```

## Loop Statements

> Statements that repeat an operation while a condition is true, such as `while` loops or `for` loops:

Reviewer: ```suggestion
[Loop](https://www.codecademy.com/resources/docs/javascript/loops) statements repeat an operation while a condition is true, such as `while` loops or `for` loops:
```

Reviewer: What about [switch](https://www.codecademy.com/resources/docs/javascript/switch)? Maybe we can just mention it and link out
